tests/blackbox/stratisd_cert.py

```diff
@@ -37,8 +37,13 @@ def make_test_pool(pool_name, pool_disks):
     (obj_path_exists, (obj_path, _)), return_code, msg = StratisDbus.pool_create(
         pool_name, pool_disks, None
     )
-    assert return_code == 0, "return_code: %s, error_msg: %s" % (return_code, msg)
-    assert obj_path_exists, "obj_path_exists: %s" % obj_path_exists
+    if return_code != 0:
```

Comment:
I think abstracting is a good idea, but we should probably abstract separately for stratisd_cert.py and stratis_cli_cert.py.

A method something like:

```
def _raise_error_exception(return_code, msg, return_value_exists):
    """
    Check result of a D-Bus call in a context where it is in error if the call fails.
    
    :param int return_code: the return code from the D-Bus call
    :param str msg: the message returned on the D-Bus
    :param bool return_value_exists: whether a value representing a valid result was returned
    """
    if return_code != 0:
       raise RuntimeError(
           "Expected return code of 0; actual return_code: %s, error_msg: %s" % (return_code, msg)
      )

    if not return_value_exists:
       raise RuntimeError("result value was default or place holder value and does not represent a valid result")
```

then invoke here as:

```
_raise_error_exception(return_code, msg, obj_path_exists)
```

would convey all the useful information.

Reply:
Resolved.